evalml/automl/engine/dask_engine.py

```diff
@@ -0,0 +1,83 @@
+import joblib
+from dask.distributed import Client
+
+from evalml.automl.engine.engine_base import (
+    EngineBase,
+    EngineComputation,
+    evaluate_pipeline,
+    score_pipeline,
+    train_pipeline
+)
+
+
+class DaskComputation(EngineComputation):
+    """A Future-like wrapper around jobs created by the DaskEngine."""
+
+    def __init__(self, dask_future):
+        self.work = dask_future
+        self.meta_data = {}
+
+    def done(self) -> bool:
+        """Is the computation done?"""
+        return self.work.done()
+
+    def get_result(self):
+        """Get the computation result.
+        Will block until the computation is finished.
+
+        Raises Exception: If computation fails. Returns traceback.
+        """
+        return self.work.result()
+
+    def cancel(self) -> None:
```

Comment:
Right now we always send out a full batch of pipelines even if we know that the search will finish before the full batch (because of `max_iterations`). In the situations where the search finishes sooner than expected, I think we should cancel the jobs that are still running. This functionality isn't in the `AutoMLSearch` loop yet though